src/Symfony/Component/HttpFoundation/Response.php

```diff
@@ -1173,7 +1119,14 @@ public function isNotFound()
      */
     public function isRedirect($location = null)
     {
-        return in_array($this->statusCode, array(201, 301, 302, 303, 307, 308)) && (null === $location ?: $location == $this->headers->get('Location'));
+        return in_array($this->statusCode, array(
+            self::HTTP_CREATED,
+            self::HTTP_MOVED_PERMANENTLY,
+            self::HTTP_FOUND,
+            self::HTTP_SEE_OTHER,
+            self::HTTP_TEMPORARY_REDIRECT,
+            self::HTTP_PERMANENTLY_REDIRECT
```

Comment:
We provide named constants because some developers asked for it.

But the meaning of all the code in Response is to deal with HTTP status codes, which are integers. The fact that we have a list of constants exposing these status codes is irrelevant for the other parts of the Response. They care about the status code itself.

Our class does not have 2 source of truth. The ``HTTP_*`` constants are **not** a source of truth here. They are a secondary asset (the source of truth is the list of integers defined in the HTTP specs)